### PR TITLE
Add `lld` symlinks to the installed directory

### PR DESCRIPTION
**Pull request by hcindyl**
Allow users to use the `lld` built from the source without updating
additional platform-specific linker names.

Also update the static_library webdoc to show how to use the linker.

### CODE REVIEW

File: iree/tools/CMakeLists.txt [hunk 1]
```diff
@@ -478,6 +478,21 @@ if(${IREE_BUILD_COMPILER})
     RUNTIME DESTINATION bin
   )
 
+  if(NOT LLD_SYMLINKS_TO_CREATE)
+  set(LLD_SYMLINKS_TO_CREATE
```
Comment (benvanik) [inline iree/tools/CMakeLists.txt:482]: can drop the darwinnew/darwinold

Comment (hcindyl) [reply iree/tools/CMakeLists.txt:482]: Will do. I copied the link set from `third_party/llvm-project/lld/tools/CMakeLists.txt`

File: iree/tools/CMakeLists.txt [hunk 2]
```diff
@@ -478,6 +478,21 @@ if(${IREE_BUILD_COMPILER})
...
+    lld-link ld.lld ld64.lld wasm-ld)
+  endif()
+
+  foreach(link ${LLD_SYMLINKS_TO_CREATE})
+    install(
+      CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink lld ${link} \
```
Comment (ScottTodd) [inline iree/tools/CMakeLists.txt:488]: What are the intended outputs and uses of this?

On Windows I get these:

```
[build] -- Up-to-date: D:/dev/projects/iree-build/install/bin/lld.exe
[build] -- Create symlink: bin/lld-link -> bin/lld
[build] -- Create symlink: bin/ld.lld -> bin/lld
[build] -- Create symlink: bin/ld64.lld -> bin/lld
[build] -- Create symlink: bin/wasm-ld -> bin/lld
```

I can't run the `lld.exe` program directly:
```
λ D:\dev\projects\iree-build\install\bin\lld.exe
lld is a generic driver.
Invoke ld.lld (Unix), ld64.lld (macOS), lld-link (Windows), wasm-ld (WebAssembly) instead
```

(the `iree-build\third_party\llvm-project\llvm\bin` directory _does_ have a `lld-link.exe` that works, but that is not installed into `iree-build/install/bin` right now)

and I might be missing something, but I can't run the symlinks either:
```
λ D:\dev\projects\iree-build\install\bin\lld-link
'D:\dev\projects\iree-build\install\bin\lld-link' is not recognized as an internal or external command,
operable program or batch file.
```

File: iree/tools/CMakeLists.txt [hunk 3]
```diff
@@ -478,6 +478,21 @@ if(${IREE_BUILD_COMPILER})
     RUNTIME DESTINATION bin
   )
 
+  if(NOT LLD_SYMLINKS_TO_CREATE)
+  set(LLD_SYMLINKS_TO_CREATE
+    lld-link ld.lld ld64.lld wasm-ld)
```
Comment (ScottTodd) [inline iree/tools/CMakeLists.txt:483]: These are different programs. Why add symlinks that group them together?

`wasm-ld` in particular has flags and behavior that `ld` doesn't: https://lld.llvm.org/WebAssembly.html

**Review by ScottTodd — CHANGES_REQUESTED** [review]

File: iree/tools/CMakeLists.txt (hunk 2 above)
Comment (ScottTodd) [reply iree/tools/CMakeLists.txt]: Can't we just use what LLVM already provides for this?

https://github.com/llvm/llvm-project/blob/main/lld/tools/lld/CMakeLists.txt
https://github.com/llvm/llvm-project/blob/main/lld/cmake/modules/AddLLD.cmake
https://github.com/llvm/llvm-project/blob/main/llvm/cmake/modules/AddLLVM.cmake

They already have the platform logic implemented, and anything custom we do here will be brittle and difficult to maintain.

Comment (hcindyl) [reply iree/tools/CMakeLists.txt]: Those macros requires `lld` to be a target built under the same directory. If I use those macros, I will get
```
TARGET 'lld' was not created in this directory.
```

**Comment by ScottTodd** [comment]
Possible overlap with https://github.com/google/iree/pull/7186?

**Comment by hcindyl** [comment]
Yes indeed. I noticed Stella's PR just this morning. We can close this one and pursuit the other path.

**Comment by stellaraccident** [comment]
> Yes indeed. I noticed Stella's PR just this morning. We can close this one and pursuit the other path.

I spent a (small) amount of time looking at how we invoke lld. I think the right/best answer is to just invoke `lld` and pass an appropriate `-flavor link|wasm|elf|...`. A couple of our call sites do that. To be robust, they probably all should. The special names/symlinks are more for interfacing as drop-in replacement with existing C toolchains.

**Comment by benvanik** [comment]
yeah all the symlinks do is let it look up progname and set -flavor, so easiest just to call in with that set (no ambiguity, no platform switches needed, etc)

**Comment by hcindyl** [comment]
Close this PR so we can move to the right approach.

**Pull request closed without merging**